onyo/lib/onyo.py

```diff
@@ -1240,3 +1241,128 @@ def unset(self, paths: Iterable[Union[Path, str]], values: str,
             self.restore()
 
         return diff
+
+    #
+    # GET
+    #
+    # TODO: This should be a staticmethod in utils
+    def _walk(self, path, depth) -> Generator:
```

Comment:
I think this connects to #259

Reply:
Filtering by depth using `Path` is easy, as `len(Path.parents)` provides the depth for each object. This can therefore also be done with the `self.assets` output directly as `assets = [asset for asset in self.assets if len(asset.parents) <= depth]`